### PR TITLE
update actions/download-/upload-artifact to v4

### PR DESCRIPTION
**Pull request by lingnoi**
Issues: #254 

# Definition of Done

The PR shall be merged only if all items mentioned in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been followed. In case an item is not applicable as described, please provide a short explanation in the description.

- [x] All steps in [CONTRIBUTING.md](https://github.com/eclipse-ankaios/ankaios/blob/main/CONTRIBUTING.md#how-to-contribute) have been handled

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -36,7 +36,7 @@ jobs:
         run: |
           mkdir -p build
           cargo about generate about.hbs > build/licenses.html
-      - uses: actions/upload-artifact@v3.1.2
+      - uses: actions/upload-artifact@v4
```
Comment (inf17101) [inline .github/workflows/build.yml:39]: I would recommend the exact stable version: v4.3.3. In the past we had some problems that the latest version was upgraded and our pipeline was crashing. This is why the documentation was extended about that: https://eclipse-ankaios.github.io/ankaios/0.3/development/ci-cd/#adding-a-new-github-action

Comment (lingnoi) [reply .github/workflows/build.yml]: ok, used: for upload-artifact v4.3.3 and v4.1.6 for download-artifact

**Comment by lingnoi** [comment]
I haves tested the workflows build.yml and release.yml by manually triggered a release https://github.com/eclipse-ankaios/ankaios/actions/runs/8816900523/job/24202338915
It shows that the updated versions `actions/download-artifact@v4.1.6` and `actions/upload-artifact@v4.3.3` are working fine:
actions_download-artifact_v4 1 6_positive
actions_upload-artifact_v4 3 3_positive

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -26,27 +26,27 @@ jobs:
     steps:
       - uses: actions/checkout@v4.1.1
       - name: Download artifacts for ankaios-linux-amd64-bin
-        uses: actions/download-artifact@v3.0.2
+        uses: actions/download-artifact@v4.1.6
```
Comment (maturar) [inline .github/workflows/release.yml:29]: The latest version of the download artifact is 4.1.7 now. Is there any reason why to use 4.1.6?

**Review by maturar — COMMENTED** [review]
I have suggested to use the latest version of the `download-artifact` action. But I am also fine with using 4.1.6.

**Review by maturar — APPROVED** [review]
LGTM :+1:

**Pull request merged**